cmd/bootstrap-post/main.go

```diff
@@ -1,80 +1,147 @@
 package main
 
 import (
-	"crypto/tls"
-	"fmt"
-	"net/http"
 	"os"
-	"tyk/tyk/bootstrap/data"
-	"tyk/tyk/bootstrap/helpers"
-	"tyk/tyk/bootstrap/readiness"
+	"strings"
+	"tyk/tyk/bootstrap/k8s"
+	"tyk/tyk/bootstrap/pkg/config"
+	"tyk/tyk/bootstrap/tyk"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	secretTypeOperator = "Operator"
+	secretTypePortal   = "Portal"
 )
 
 func main() {
-	err := data.InitPostInstall()
+	log := logrus.New()
+
+	conf, err := config.NewConfig()
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
 	}
 
-	err = readiness.CheckIfRequiredDeploymentsAreReady()
+	level, err := logrus.ParseLevel(conf.Log)
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
+		log.Infof(
+			"Failed to parse log level %v, continuing with the default log level Info, err %v", conf.Log, err,
+		)
 
-	tp := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: data.BootstrapConf.InsecureSkipVerify},
+		level = logrus.InfoLevel
 	}
-	client := http.Client{Transport: tp}
 
-	fmt.Println("Started creating dashboard org")
+	log.SetLevel(level)
+	log.WithField("level", level.String()).Info("Set the log level")
 
-	err = helpers.CheckForExistingOrganisation(client)
+	k8sClient, err := k8s.NewClient(conf, log.WithField("Client", "Kubernetes"))
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
+	}
+
+	if err = k8sClient.CheckIfRequiredDeploymentsAreReady(); err != nil {
+		exit(log, err)
 	}
 
-	fmt.Println("Finished creating dashboard org")
-	fmt.Println("Generating dashboard credentials")
+	if conf.BootstrapDashboard {
+		conf.K8s.DashboardSvcUrl, err = k8sClient.DiscoverDashboardSvc()
+		if err != nil {
+			exit(log, err)
+		}
+	}
 
-	err = helpers.GenerateDashboardCredentials(client)
+	tykSvc := tyk.NewClient(conf, log.WithField("Client", "Tyk"))
+
+	orgExists, err := tykSvc.OrgExists()
```

Comment:
We should not check this either if bootstrapping for dashboard is disabled.

Reply:
updated in 7e0c8e1e841ad97a9aff4c60a3d313850916e451